service/notification/NotificationProcessor.php

```diff
@@ -335,4 +351,32 @@ private function isCurrentOrderStatusANonFinalStatus($currentOrderStatus)
 
         return false;
     }
+
+    /**
+     * @param $notification
+     * @param OrderCore $order
+     * @return bool
+     * @throws \Exception
+     */
+    private function validateWithCartAndOrder($notification, OrderCore $order)
+    {
+        $cart = \Cart::getCartByOrderId($order->id);
```

Comment:
Whenever you work with prestashop core objects you can validate thembefore using them with the \Validate::isLoadedObject() function